src/dataset.jl

```diff
@@ -178,7 +178,71 @@ function write(
     filename::AbstractString;
     kwargs...,
 )::Nothing
-    destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    drivername = GDAL.gdalgetdrivershortname(getdriver(dataset).ptr)
+    if drivername in gdal_raster_drivers
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    elseif drivername in gdal_vector_drivers
+        writevectords(dataset, filename; kwargs...)
+    else
+        error("Dataset format not recognized.")
+    end
+    return nothing
+end
+
+"""
+    writevectords(ds, filename; driver=ArchGDAL.getdriver(ds), options::Vector{String}=[""], chunksize=20000)
+
+Writes the vector dataset to the designated filename. The options are passed to the newly created dataset and
+have to be given as a list of strings in KEY=VALUE format. The chunksize controls the number of features written 
+in each database transaction, e.g. for SQLite.
+"""
+function writevectords(ds, filename; driver=getdriver(ds), options::Vector{String}=[""], chunksize=20000)
+    create(filename; driver=driver, options=options) do target
+        for layeridx in 0:nlayer(ds)-1
+            sourcelayer = getlayer(ds, layeridx)
+            sourcelayerdef = layerdefn(sourcelayer)
+            createlayer(name=getname(sourcelayer),
+                                 dataset=target,
+                                 geom=wkbUnknown,  # GPKG only works when using unknown here, deleting the field and recreating it later
+                                 spatialref=getspatialref(sourcelayer)) do targetlayer
+                targetlayerdefn = layerdefn(targetlayer)
+                
+                # the following try catch block is a bit hacky
+                # geopackage files dont work with the default geom field definition,
+                # because the driver expects a field "GEOMETRY" but the default is "Geometry"!?
+                # on the other hand, the ESRI Shapefile driver does not support GDAL.ogr_l_creategeomfield
+                # hence the try catch block as a workaround; it seems to work even after having delted the geomdefn
```

Comment:
Hahaha this is so annoying, thank you for the investigation and context!

Reply:
Do you know where the difference in field names for GPKG comes from? Does ArchGDAL set any field names to some default values? Probably not, I gues...

Reply:
> Do you know where the difference in field names for GPKG comes from? Does ArchGDAL set any field names to some default values?

I'm afraid I don't know.. ArchGDAL does not set field names to default values

Reply:
There seems to be a requirement in geopackage:

> The `geometry_type_name` value in a `gpkg_geometry_columns` row SHALL be one of the uppercase geometry type names specified in [Geometry Types (Normative)](https://www.geopackage.org/spec/#geometry_types).

(source: https://www.geopackage.org/spec/#_gpkg_geometry_columns)